event/event.go

```diff
@@ -0,0 +1,208 @@
+	// Pressure is the pressure that is applied to the gamepad button.
+	// It varies between 0.0 for not pressed, and 1.0 for completely pressed.
+	Pressure float32
+}
+
+// GamepadButtonDown is a gamepad button release event.
+// The data is identical to a GamePadButtonDown event.
+type GamepadButtonUp GamepadButtonDown
+
+// GamepadAttach happens when a new gamepad is attached.
+type GamepadAttach struct {
+	// ID represents which gamepad caused the event.
+	ID int
+	// Axes represents the amount of axes the gamepad has.
+	Axes int
+	// Buttons represents the amount of buttons the gamepad has.
+	Buttons int
+}
+
+// GamepadDetach happens when a gamepad is detached.
+type GamepadDetach struct {
+	// ID represents which gamepad caused the event.
+	ID int
+}
+
+// MouseMove is a mouse movement event.
+type MouseMove struct {
+	// X is the X position of the mouse pointer.
+	X float32
+	// Y is the Y position of the mouse pointer.
+	Y float32
+	// DeltaX is the change in X since the last MouseMove event.
+	DeltaX float32
+	// DeltaY is the change in Y since the last MouseMove event.
+	DeltaY float32
+}
+
+// MouseWheel is a mouse wheel event.
+type MouseWheel struct {
+	// X is the X position of the mouse wheel.
+	X float32
+	// Y is the Y position of the mouse wheel.
+	Y float32
+	// DeltaX is the change in X since the last MouseWheel event.
+	DeltaX float32
+	// DeltaY is the change in Y since the last MouseWheel event.
+	DeltaY float32
+}
+
+// MouseButtonDown is a mouse button press event.
+type MouseButtonDown struct {
+	// X is the X position of the mouse pointer.
+	X float32
+	// Y is the Y position of the mouse pointer.
+	Y float32
+	// Button is the button on the mouse that was pressed.
+	// TODO: this should change later from an int to an enumeration type
+	Button int
+	// Pressure is the pressure applied on the mouse button.
+	// It varies between 0.0 for not pressed, and 1.0 for completely pressed.
+	Pressure float32
+}
+
+// MouseButtonDown is a mouse button Release event.
+// The data is identical to a MouseButtonDown event.
+type MouseButtonUp MouseButtonDown
+
+// MouseEnter occurs when the mouse enters the view window.
+type MouseEnter struct {
+	// X is the X position of the mouse pointer.
+	X float32
+	// Y is the Y position of the mouse pointer.
+	Y float32
+}
+
+// MouseLeave occurs when the mouse leaves the view window.
+// The data is identical to MouseEnter
+type MouseLeave MouseEnter
+
+// ViewUpdate occurs when the application is ready to update
+// the next frame on the view port.
+type ViewUpdate struct {
+	// No data neccesary, for now.
+}
+
+// ViewSize occurs when the size of the application's view port changes.
+type ViewSize struct {
+	// Width is the actual, real width of the view.
+	Width float32
+	// Height is the actual, real height of the view.
+	Height float32
+	// X is the X position of the view on the physical screen.
+	X float32
+	// Y is the Y position of the view on the physical screen.
+	Y float32
```

Comment:
* Use `int` instead of `float32`.
* Add explanation that the unit is device-independent pixel.
* X, Y are not needed so far (Ebiten does not care these values)

Reply:
Ok, for the screen, int is fine since this is in device independent pixels which are always a whole number for the screen. I will remove the X and Y. since they are not used yet by ebiten.